include/gevolution/Particles_gevolution.hpp

```diff
@@ -94,16 +99,211 @@ class Particles_gevolution :
         particle_dataType>
 {
   public:
+    
     using value_type = particle; // helper attribute used in containers for metaprogramming
 
     void saveGadget2 (std::string filename, gadget2_header &hdr,
                       const int tracer_factor = 1) const ;
-    void saveGadget2 (std::string filename, gadget2_header &hdr,
-                      lightcone_geometry &lightcone, double dist, double dtau,
-                      double dtau_old, double dadtau,
-                      double vertex[MAX_INTERSECTS][3], const int vertexcount,
-                      std::set<long> &IDbacklog, std::set<long> &IDprelog,
-                      Field<Real> *phi, const int tracer_factor = 1);
+   
+    template<typename selector_type>
+    void saveGadget2 (
+        const std::string filename, 
+        gadget2_header hdr,
+        selector_type select_function,
```

Comment:
Comment here. This function, unlike the previous `saveGadget2`, writes only the selected particles to file.

Reply:
ok f5919b4ba63b8131c18b603fefd2a70d82730fea